src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -100,49 +100,6 @@ namespace Antares
 {
 namespace Data
 {
-Data::ThermalCluster::ThermalCluster(Area* parent, uint nbParallelYears) :
- Cluster(parent),
- groupID(thermalDispatchGrpOther1),
- mustrun(false),
- mustrunOrigin(false),
- nominalCapacityWithSpinning(0.),
- minStablePower(0.),
- minUpTime(1),
- minDownTime(1),
- spinning(0.),
- forcedVolatility(0.),
- plannedVolatility(0.),
- forcedLaw(thermalLawUniform),
- plannedLaw(thermalLawUniform),
- marginalCost(0.),
- spreadCost(0.),
- fixedCost(0.),
- startupCost(0.),
- marketBidCost(0.),
- groupMinCount(0),
- groupMaxCount(0),
- annuityInvestment(0),
- PthetaInf(HOURS_PER_YEAR, 0),
- prepro(nullptr),
- productionCost(nullptr),
- unitCountLastHour(nullptr),
- productionLastHour(nullptr),
- pminOfAGroup(nullptr)
-{
-    // assert
-    assert(parent and "A parent for a thermal dispatchable cluster can not be null");
-
-    unitCountLastHour = new uint[nbParallelYears];
-    productionLastHour = new double[nbParallelYears];
-    pminOfAGroup = new double[nbParallelYears];
-    for (uint numSpace = 0; numSpace < nbParallelYears; ++numSpace)
-    {
-        unitCountLastHour[numSpace] = 0;
-        productionLastHour[numSpace] = 0.;
-        pminOfAGroup[numSpace] = 0.;
```

Comment:
Here initialization of **unitCountLastHour**, **productionLastHour** and **pminOfAGroup**

Reply:
Those var were double/uint arrays before, like `double* productionLastHour;`
We replaced those by vectors and resize allows to set them to the right size and filling them with zeros